File: Api/Models/JobConfiguration.cs
namespace GearForgeXI.Models;

/// <summary>
/// Database-driven job configuration containing all job-specific data
/// </summary>
public class JobConfiguration
{
    public int Id { get; set; }

    /// <summary>
    /// The job this configuration applies to
    /// </summary>
    public int JobId { get; set; }
    public Job Job { get; set; } = null!;

    /// <summary>
    /// Base stats at level 99 for this job
    /// </summary>
    public List<JobBaseStat> JobBaseStats { get; set; } = new();

    /// <summary>
    /// Job traits that are learned at specific levels
    /// </summary>
    public List<JobTrait> JobTraits { get; set; } = new();

    /// <summary>
    /// Stat bonuses gained from job point categories when mastered
    /// </summary>
    public List<JobPointBonus> JobPointBonuses { get; set; } = new();

    /// <summary>
    /// Stat bonuses gained from master levels
    /// </summary>
    public List<MasterLevelBonus> MasterLevelBonuses { get; set; } = new();
}

File: Client/src/models/JobConfiguration.ts
import type { Job } from "./Job";
import type { Stat } from "./Stat";

export interface JobConfiguration {
  id: number;
  jobId: number;
  job: Job;
  jobBaseStats: JobBaseStat[];
  jobBaseSkills: JobBaseSkill[];
  jobTraits: JobTrait[];
  jobPointBonuses: JobPointBonus[];
  jobGifts: JobGift[];
  masterLevelBonuses: MasterLevelBonus[];
}

export interface JobBaseStat {
  id: number;
  jobConfigurationId: number;
  statId: number;
  stat: Stat;
  value: number;
}

export interface JobBaseSkill {
  id: number;
  jobConfigurationId: number;
  statId: number;
  stat: Stat;
  value: number;
}

export interface JobTrait {
  id: number;
  jobConfigurationId: number;
  name: string;
  level: number;
  statId: number;
  stat: Stat;
  value: number;
}

export interface JobPointBonus {
  id: number;
  jobConfigurationId: number;
  statId: number;
  stat: Stat;
  value: number;
}

export interface JobGift {
  id: number;
  jobConfigurationId: number;
  statId: number;
  stat: Stat;
  value: number;
}

export interface MasterLevelBonus {
  id: number;
  jobConfigurationId: number;
  level: number;
  statId: number;
  stat: Stat;
  value: number;
}
